application/aam-backend-service/src/main/kotlin/com/aamdigital/aambackendservice/reporting/reportcalculation/core/DefaultCreateReportCalculationUseCase.kt

```diff
@@ -21,9 +21,26 @@ class DefaultCreateReportCalculationUseCase(
             args = request.args
         )
 
-        return reportingStorage.storeCalculation(calculation)
-            .map {
-                handleResponse(it)
+        return reportingStorage.fetchCalculations(request.report)
+            .flatMap { reportCalculations ->
+                val i = reportCalculations.filter { reportCalculation ->
+                    reportCalculation.status == ReportCalculationStatus.PENDING &&
+                            reportCalculation.args == calculation.args
+                }
+
+                if (i.isNotEmpty()) {
+                    Mono.just(
+                        CreateReportCalculationResult.Success(
```

Comment:
Does this mean if a pending calculation is found, we immediately return a success calculation here? 
And if so, us that correct behavior - or should this still be a "Pending" result somehow?